Cannot find module './injectIntl' or its corresponding type declarations

**Which package?**

`packages/react-intl`, version `@4.6.10`.

**Describe the bug**

When compiling a TypeScript project that uses `useIntl` and `defineMessages`, the compilation fails as follows:

```sh
node_modules/react-intl/lib/react-intl.d.ts:95:57 - error TS2307: Cannot find module './injectIntl' or its corresponding type declarations.

95 export declare const FormattedPlural: React_2.FC<import("./injectIntl").WithIntlProps<Props_2>> & {
                                                           ~~~~~~~~~~~~~~
```

**To Reproduce**
Steps to reproduce the behavior:

1. Any TypeScript projects
2. Run `tsc` to compile the project

**Expected behavior**
No error occurs, projects get compiled successfully.

**Desktop (please complete the following information):**

- OS: [Mac OS v10.15.5]
- NodeJS [v12.16.2]
- react-intl Version [4.6.10]

**Additional context**

Version `4.6.9` doesn't have this issue.
After some digging, I'm guessing the issue was introduced by the `4.6.10` release(2 days ago) that adds children typing to `injectIntl`, the generated `react-intl.d.ts` includes an `import('./injectIntl')` statement, but the`./injectIntl` module stays in `./components/` directory.

--- a/package.json
+++ b/package.json
@@ -25,7 +25,7 @@
     "test": "yarn run prettier:check && lerna run test"
   },
   "devDependencies": {
-    "@microsoft/api-extractor": "^7.7.13",
+    "@microsoft/api-extractor": "^7.8.12",
     "@rollup/plugin-json": "^4.0.3",
     "@types/jest": "^25.2.1",
     "@typescript-eslint/eslint-plugin": "^2.29.0",

--- a/packages/intl-numberformat/src/core.ts
+++ b/packages/intl-numberformat/src/core.ts
@@ -21,11 +21,15 @@ import {names as numberingSystemNames} from './data/numbering-systems.json';
 import formatToParts from './format_to_parts';
 // eslint-disable-next-line import/no-cycle
 import getInternalSlots from './get_internal_slots';
-import {NumberFormatConstructor, NumberFormatOptions} from './types';
+import {
+  NumberFormatConstructor,
+  NumberFormatOptions,
+  NumberFormat as NumberFormatType,
+} from './types';
 import type {getCanonicalLocales} from '@formatjs/intl-getcanonicallocales';
 
 // Merge declaration with the constructor defined below.
-export type NumberFormat = import('./types').NumberFormat;
+export type NumberFormat = NumberFormatType;
 
 const VALID_NUMBERING_SYSTEM_NAMES = new Set(numberingSystemNames);
 

--- a/packages/react-intl/package.json
+++ b/packages/react-intl/package.json
@@ -168,8 +168,7 @@
   ],
   "devDependencies": {
     "@formatjs/intl-pluralrules": "^2.2.9",
-    "@microsoft/api-documenter": "^7.7.18",
-    "@microsoft/api-extractor": "^7.7.13",
+    "@microsoft/api-documenter": "^7.8.14",
     "@types/benchmark": "^1.0.31",
     "@types/enzyme": "^3.10.5",
     "@types/jest": "^25.2.1",

--- a/packages/react-intl/src/components/plural.tsx
+++ b/packages/react-intl/src/components/plural.tsx
@@ -47,6 +47,10 @@ FormattedPlural.defaultProps = {
 
 FormattedPlural.displayName = 'FormattedPlural';
 
-export default withIntl(FormattedPlural) as React.FC<WithIntlProps<Props>> & {
+// Explicitly annotate type here to workaround API extractor's inability to handle `import('./someModule')`
+// type annotations when rolling up DTS file.
+const FormattedPluralWithIntl: React.FC<WithIntlProps<Props>> & {
   WrappedComponent: React.ComponentType<Props>;
-};
+} = withIntl(FormattedPlural);
+
+export default FormattedPluralWithIntl;

--- a/packages/react-intl/src/index.ts
+++ b/packages/react-intl/src/index.ts
@@ -15,7 +15,7 @@ import {
   createFormattedComponent,
   createFormattedDateTimePartsComponent,
 } from './components/createFormattedComponent';
-import {CustomFormatConfig} from './types';
+import {CustomFormatConfig, FormatDateOptions} from './types';
 import {NumberFormatOptions} from '@formatjs/intl-numberformat';
 import {IntlListFormatOptions} from '@formatjs/intl-listformat';
 import {DisplayNamesOptions} from '@formatjs/intl-displaynames/lib';
@@ -58,12 +58,19 @@ export const FormattedDisplayName: React.FC<
     value: string | number | object;
   }
 > = createFormattedComponent('formatDisplayName');
-export const FormattedDateParts = createFormattedDateTimePartsComponent(
-  'formatDate'
-);
-export const FormattedTimeParts = createFormattedDateTimePartsComponent(
-  'formatTime'
-);
+export const FormattedDateParts: React.FC<
+  FormatDateOptions & {
+    value: Parameters<Intl.DateTimeFormat['format']>[0] | string;
+    children(val: Intl.DateTimeFormatPart[]): React.ReactElement | null;
+  }
+> = createFormattedDateTimePartsComponent('formatDate');
+export const FormattedTimeParts: React.FC<
+  FormatDateOptions & {
+    value: Parameters<Intl.DateTimeFormat['format']>[0] | string;
+    children(val: Intl.DateTimeFormatPart[]): React.ReactElement | null;
+  }
+> = createFormattedDateTimePartsComponent('formatTime');
+
 export {FormattedNumberParts} from './components/createFormattedComponent';
 export {default as FormattedRelativeTime} from './components/relative';
 export {default as FormattedPlural} from './components/plural';

